wgpu-hal/src/dx12/adapter.rs

```diff
@@ -270,7 +271,7 @@ impl super::Adapter {
                         .min(crate::MAX_VERTEX_BUFFERS as u32),
                     max_vertex_attributes: d3d12::D3D12_IA_VERTEX_INPUT_RESOURCE_SLOT_COUNT,
                     max_vertex_buffer_array_stride: d3d12::D3D12_SO_BUFFER_MAX_STRIDE_IN_BYTES,
-                    max_push_constant_size: 0,
+                    max_push_constant_size: 128,
```

Comment:
wouldn't this technically be 256 - 12?

Reply:
I'll be honest, I copied this from the `wgpu-hal` code

Reply:
Actually, I've took some time to read about `dx12` root signatures and it's limits and we can't make the `max_push_constant_size` that big.

That's because the root signature has a limit of `64 DWORDS` (`256 bytes`):
- push constants consume a `DWORD` for each `4 bytes` of data (1:1)
- If a bind group has buffers it will consume a `DWORD` for the descriptor table
- If a bind group has samplers it will consume a `DWORD` for the descriptor table
- Each dynamic buffer will consume `2 DWORDs` for the root descriptor
- The special constants count as constants and occupy `3 DWORDS`

Source: https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signature-limits#memory-limits-and-costs

Since we can't know beforehand all root signatures that will be created, we should set a more reasonable `max_push_constant_size` and I think that half (`32 DWORDS = 128 bytes`) like `gfx-hal` had is a reasonable value.

Reply:
That seems fair enough - we should probably document this in the code

Reply:
done